x-pack/plugins/security_solution/server/lib/detection_engine/signals/utils.ts

```diff
@@ -661,3 +663,356 @@ export const createTotalHitsFromSearchResult = ({
       : searchResult.hits.total.value;
   return totalHits;
 };
+
+export interface TimestampsAndIndices {
+  [timestampString: string]: string[]; // maps timestampString like @timestamp or 'event.ingested' to the indices that contain that timestamp mapping
+}
+
+export const checkIndexMappingsForTimestampFields = async (
+  indices: string[],
+  timestamps: string[],
+  services: AlertServices,
+  logger: Logger,
+  buildRuleMessage: BuildRuleMessage
+): Promise<TimestampsAndIndices> => {
+  try {
+    const foundMappings: GetFieldMappingType = await services.callCluster(
```

Comment:
As in `findIndicesWithTimestampAndWithout`, maintaining the association between index pattern and concrete index here would be useful for code that builds on this function.

Solution we discussed is to request the field mappings per index pattern.